protocol/x/revshare/keeper/revshare.go

```diff
@@ -32,3 +36,49 @@ func (k Keeper) GetMarketMapperRevenueShareParams(
 	k.cdc.MustUnmarshal(b, &params)
 	return params
 }
+
+// Function to serialize market mapper rev share params and store in the
+// module store
+func (k Keeper) SetMarketMapperRevShareDetails(
+	ctx sdk.Context,
+	marketId uint32,
+	params types.MarketMapperRevShareDetails,
+) (err error) {
+	// Store the rev share details for provided market in module store
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.MarketMapperRevSharePrefix))
+	b := k.cdc.MustMarshal(&params)
+	store.Set(lib.Uint32ToKey(marketId), b)
+	return nil
+}
+
+// Function to retrieve marketmapper revshare params from module store
+func (k Keeper) GetMarketMapperRevShareDetails(
+	ctx sdk.Context,
+	marketId uint32,
+) (params types.MarketMapperRevShareDetails, err error) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.MarketMapperRevSharePrefix))
+	b := store.Get(lib.Uint32ToKey(marketId))
+	if b == nil {
+		return params, fmt.Errorf("MarketMapperRevShareDetails not found for marketId: %d", marketId)
+	}
+	k.cdc.MustUnmarshal(b, &params)
+	return params, nil
+}
+
+// Function to perform all market creation actions for the revshare module
+func (k Keeper) CreateNewMarketRevShare(ctx sdk.Context, marketId uint32) (err error) {
+	revShareParams := k.GetMarketMapperRevenueShareParams(ctx)
+
+	validDurationSeconds := int64(revShareParams.ValidDays * 24 * 60 * 60)
```

Comment:
nit: uint64?

Reply:
Unix time is int64 so this needs to be the same for the addition. The result later gets converted to uint64 which is fine because its not possible for these to be negative